code/modules/reagents/chemistry/reagents/medicine_reagents.dm

```diff
@@ -503,10 +500,24 @@
 
 /datum/reagent/medicine/ephedrine/on_mob_life(mob/living/M)
 	M.status_flags |= GOTTAGOFAST
-	M.AdjustParalysis(-1, 0)
-	M.AdjustStunned(-1, 0)
-	M.AdjustWeakened(-1, 0)
-	M.adjustStaminaLoss(-1*REM, 0)
+	if(M.reagents.has_reagent("nutriment"))
+		M.reagents.remove_reagent("nutriment", 5)
+	if(M.reagents.has_reagent("vitamin"))
+		M.reagents.remove_reagent("vitamin", 5)
+	if(prob(34))
+		if(M.nutrition > 15)
+			M.nutrition -= rand(5, 15) //Lose nutrition very rapidly.
+		else
+			if(M.nutrition > 0)//If you are nearly out of nutrition it slows down a bit but you could go all the way down. this prevents negatives.
+				M.nutrition -= 1
+
+	if(M.reagents.has_reagent("methamphetamine"))
```

Comment:
Also this does not purge people of said chems, so sure a guy is on fire but now he is running fast and has anti-stuns.